prowler/providers/azure/azure_provider.py

```diff
@@ -244,23 +258,25 @@ def validate_arguments(
             SystemExit: If browser authentication is enabled but the tenant ID is not provided.
             SystemExit: If tenant ID is provided but browser authentication is not enabled.
         """
-        if (
+        if not browser_auth and tenant_id:
+            raise AzureTenantIDNoBrowserAuthError(
+                file=os.path.basename(__file__),
+                original_exception="Azure Tenant ID (--tenant-id) is required only for browser authentication mode",
+            )
+        elif (
             not az_cli_auth
             and not sp_env_auth
             and not browser_auth
             and not managed_identity_auth
         ):
-            raise SystemExit(
-                "Azure provider requires at least one authentication method set: [--az-cli-auth | --sp-env-auth | --browser-auth | --managed-identity-auth]"
+            raise AzureNoAuthenticationMethodError(
+                file=os.path.basename(__file__),
+                original_exception="Azure provider requires at least one authentication method set: [--az-cli-auth | --sp-env-auth | --browser-auth | --managed-identity-auth]",
```

Comment:
Since there is no original exception, that line is just the exception we generate.

This comment applies to the rest of the file.

Reply:
I add this in order to get more info about why we are raising the exception, I can just to delete it but we won't get info.

Reply:
But I meant to change `original_exception` to `message` since there is no original exception.